# There should be only one class per file (only-one-class)

Multiple classes inside of one file are not allowed.

## Rule Details

This rule aims to...

Examples of **incorrect** code for this rule:

```js
class A {
    /** ... */
}

class B {
    /** ... */
}
```

Examples of **correct** code for this rule:

```js
class A {
    /** ... */
}
```
